documentation/book/con-the-upgrade-process.adoc

```diff
@@ -0,0 +1,62 @@
+// This module is included in the following assemblies:
+//
+// assembly-upgrading-kafka-versions.adoc
+
+[id='con-the-upgrade-process-{context}']
+= The upgrade process
+
+A given version of the Cluster Operator knows certain details about a given set of Kafka versions. 
```

Comment:
CHANGE

A given version of the Cluster Operator knows certain details about a given set of Kafka versions. 
For the purpose of a Kafka upgrade the important attributes are:

* The version of the Kafka protocol to be used between brokers. (this is set via the `interbroker.protocol.version` Kafka configuration parameter),
* The format version to be used for messages stored in the Kafka log (this is set via the `log.message.format.version` Kafka configuration parameter),
* The required version of Zookeeper.

The differences in these attributes between the Kafka version being upgraded from and the Kafka version being upgraded to will determine exactly how the upgrade process works.

TO

When upgrading Kafka, there are three attributes known to the Cluster Operator that will determine exactly how the upgrade process will be performed.

* `interbroker.protocol.version`:: The version of the Kafka protocol to be used between brokers. 
* `log.message.format.version`:: The format version to be used for messages stored in the Kafka log.
* `WHAT IS THE ZOOKEEPER VERSION PARAMETER`:: The required version of Zookeeper.

Reply:
That's not getting at what I'm trying to say. The point is that there _is_ a protocol for communication between Kafka brokers, that it has different versions and that if the different broker versions support different versions of the protocol then we need to do stuff to upgrade that communication. We do that via the `interbroker.protocol.version` setting, but emphasizing the parameter is not the point, it's just the mechanism. When I used the word "attribute" I dodn't mean "configuration parameter", I meant a "property" (but not in the configuration sense).

Likewise the the message version. The configuration parameter is not the point. The point is that any difference in the message format between the different broker versions is a consideration for how the upgrade will work.

Finally, to answer your question about Zookeeper, there is no config parameter for the ZK version. Again, the point is that if the brokers needed different versions of ZK then it would also be a factor in how the upgrade would work.